File: trunk/tests/CoreTests/UsingResources/UsingResources.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SharpKit.JavaScript;

namespace CoreTests.UsingResources
{
    [JsType(JsMode.Prototype)]
    class UsingResources
    {
        void Foo()
        {
            using (Moo())
            {
            }
            using (Moo())
                Console.WriteLine();
            using (IDisposable x = Moo(), y = Bar())
            {
            }
        }

        private IDisposable Moo()
        {
            throw new NotImplementedException();
        }

        private IDisposable Bar()
        {
            throw new NotImplementedException();
        }
    }
}

File: tests/CoreTests/UsingResources/UsingResources.js
;
if (typeof(CoreTests) == "undefined")
    var CoreTests = {};
if (typeof(CoreTests.UsingResources) == "undefined")
    CoreTests.UsingResources = {};
CoreTests.UsingResources.UsingResources = function (){
};
CoreTests.UsingResources.UsingResources.prototype.Foo = function (){
    var $r1 = this.Moo();
    try{
    }
    finally{
        $r1.Dispose();
    }
    var $r2 = this.Moo();
    try{
        System.Console.WriteLine();
    }
    finally{
        $r2.Dispose();
    }
    var x = this.Moo(),y = this.Bar();
    try{
    }
    finally{
        x.Dispose();
        y.Dispose();
    }
};
CoreTests.UsingResources.UsingResources.prototype.Moo = function (){
    throw $CreateException(new System.NotImplementedException.ctor(), new Error());
};
CoreTests.UsingResources.UsingResources.prototype.Bar = function (){
    throw $CreateException(new System.NotImplementedException.ctor(), new Error());
};

